From e725ce74537766bd0f6492a3b4bd7b25c54bf7ff Mon Sep 17 00:00:00 2001
From: Evoke Zhang <evoke.zhang@amlogic.com>
Date: Mon, 21 Jul 2014 10:08:53 +0800
Subject: [PATCH 4547/5965] PD#94534: m8/m8m2: vpu: update vpu clock table for
 clock_muxing changed

Squashed commit of the following:

commit 38bd366f7b5c3ffa84a77bdc0dd09af628aaef6a
Author: Evoke Zhang <evoke.zhang@amlogic.com>
Date:   Fri Jul 18 13:47:11 2014 +0800

    PD#94534: m8/m8m2: dtd: update vpu clk table in dtd

commit 7eb9f4077f1d605fbca4b50609f75f57af848df0
Author: Evoke Zhang <evoke.zhang@amlogic.com>
Date:   Fri Jul 18 12:47:49 2014 +0800

    PD#94534: m8/m8m2: vpu: update vpu clock table for clock_muxing changed
---
 arch/arm/boot/dts/amlogic/meson8_ZeBu.dtd     | 14 ++--
 arch/arm/boot/dts/amlogic/meson8_k100_1G.dtd  | 14 ++--
 arch/arm/boot/dts/amlogic/meson8_k100_2G.dtd  | 14 ++--
 arch/arm/boot/dts/amlogic/meson8_k101_1G.dtd  | 14 ++--
 .../arm/boot/dts/amlogic/meson8_k101_512M.dtd | 14 ++--
 arch/arm/boot/dts/amlogic/meson8_k102_v1.dtd  | 14 ++--
 arch/arm/boot/dts/amlogic/meson8_k150_v1.dtd  | 14 ++--
 arch/arm/boot/dts/amlogic/meson8_k160_v1.dtd  | 14 ++--
 arch/arm/boot/dts/amlogic/meson8_k200a_1G.dtd | 14 ++--
 arch/arm/boot/dts/amlogic/meson8_k200a_2G.dtd | 14 ++--
 .../boot/dts/amlogic/meson8_k200a_2G_4k2k.dtd | 14 ++--
 .../dts/amlogic/meson8_k200a_2G_hdmiin.dtd    | 14 ++--
 .../dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd | 14 ++--
 .../dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd | 14 ++--
 arch/arm/boot/dts/amlogic/meson8_k200b_2G.dtd | 14 ++--
 .../boot/dts/amlogic/meson8_k200b_2G_4k2k.dtd | 14 ++--
 .../dts/amlogic/meson8_k200b_2G_hdmiin.dtd    | 14 ++--
 arch/arm/boot/dts/amlogic/meson8_pxp.dtd      | 14 ++--
 arch/arm/boot/dts/amlogic/meson8_skt.dtd      | 14 ++--
 .../arm/boot/dts/amlogic/meson8m2_n100_1G.dtd | 14 ++--
 .../boot/dts/amlogic/meson8m2_n101_512M.dtd   | 14 ++--
 .../arm/boot/dts/amlogic/meson8m2_n200_2G.dtd | 14 ++--
 arch/arm/boot/dts/amlogic/meson8m2_pxp.dtd    | 14 ++--
 arch/arm/boot/dts/amlogic/meson8m2_skt.dtd    | 14 ++--
 arch/arm/mach-meson8/vpu.c                    | 69 +++++++++++++------
 25 files changed, 216 insertions(+), 189 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/meson8_ZeBu.dtd b/arch/arm/boot/dts/amlogic/meson8_ZeBu.dtd
index 9a57330112db..35856694f9f5 100755
--- a/arch/arm/boot/dts/amlogic/meson8_ZeBu.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_ZeBu.dtd
@@ -100,13 +100,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <4>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <3>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// ***************************************************************************************
diff --git a/arch/arm/boot/dts/amlogic/meson8_k100_1G.dtd b/arch/arm/boot/dts/amlogic/meson8_k100_1G.dtd
index 4ba020f2650d..a2f82edace83 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k100_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k100_1G.dtd
@@ -116,13 +116,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_k100_2G.dtd b/arch/arm/boot/dts/amlogic/meson8_k100_2G.dtd
index ced5e2f58707..ebf37a2e60ac 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k100_2G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k100_2G.dtd
@@ -116,13 +116,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_k101_1G.dtd b/arch/arm/boot/dts/amlogic/meson8_k101_1G.dtd
index 7f2c6f1768f8..bf6ae14cfd67 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k101_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k101_1G.dtd
@@ -115,13 +115,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <4>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <3>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_k101_512M.dtd b/arch/arm/boot/dts/amlogic/meson8_k101_512M.dtd
index b589e321200c..2be5f1ff1321 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k101_512M.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k101_512M.dtd
@@ -116,13 +116,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <4>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <3>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_k102_v1.dtd b/arch/arm/boot/dts/amlogic/meson8_k102_v1.dtd
index 124bfef8dc29..d8149fd3e2b0 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k102_v1.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k102_v1.dtd
@@ -110,13 +110,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <4>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <3>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_k150_v1.dtd b/arch/arm/boot/dts/amlogic/meson8_k150_v1.dtd
index cee1b6a65895..35cd709c8b08 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k150_v1.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k150_v1.dtd
@@ -109,13 +109,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <4>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <3>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_k160_v1.dtd b/arch/arm/boot/dts/amlogic/meson8_k160_v1.dtd
index 83ac85281a3a..eca47e5e071b 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k160_v1.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k160_v1.dtd
@@ -109,13 +109,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <4>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <3>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200a_1G.dtd b/arch/arm/boot/dts/amlogic/meson8_k200a_1G.dtd
index 159bb2ef4ad0..58746b3bc3bb 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k200a_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200a_1G.dtd
@@ -118,13 +118,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200a_2G.dtd b/arch/arm/boot/dts/amlogic/meson8_k200a_2G.dtd
index 85b6fa877ef1..603f69c3127e 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k200a_2G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200a_2G.dtd
@@ -111,13 +111,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200a_2G_4k2k.dtd b/arch/arm/boot/dts/amlogic/meson8_k200a_2G_4k2k.dtd
index 616aec7e10d3..11a92cf0dde4 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k200a_2G_4k2k.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200a_2G_4k2k.dtd
@@ -111,13 +111,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200a_2G_hdmiin.dtd b/arch/arm/boot/dts/amlogic/meson8_k200a_2G_hdmiin.dtd
index 9c9a650db352..34d43d357fc3 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k200a_2G_hdmiin.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200a_2G_hdmiin.dtd
@@ -111,13 +111,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd
index b2e17d4812b6..5733b9fda415 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd
@@ -120,13 +120,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd
index 27a9448e87cc..315cc261b4f3 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd
@@ -120,13 +120,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_2G.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_2G.dtd
index 3ebbf3976f0b..3682d53f74ef 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_2G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_2G.dtd
@@ -110,13 +110,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_4k2k.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_4k2k.dtd
index fb0f0336ced0..0270928670e1 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_4k2k.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_4k2k.dtd
@@ -110,13 +110,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_hdmiin.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_hdmiin.dtd
index 1803a7c26a03..1f2c0bdc53ea 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_hdmiin.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_hdmiin.dtd
@@ -110,13 +110,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_pxp.dtd b/arch/arm/boot/dts/amlogic/meson8_pxp.dtd
index e6104d5302d3..3b09cef2d9d1 100755
--- a/arch/arm/boot/dts/amlogic/meson8_pxp.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_pxp.dtd
@@ -110,13 +110,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8_skt.dtd b/arch/arm/boot/dts/amlogic/meson8_skt.dtd
index f0510289ecd6..989446259e73 100755
--- a/arch/arm/boot/dts/amlogic/meson8_skt.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_skt.dtd
@@ -110,13 +110,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8m2_n100_1G.dtd b/arch/arm/boot/dts/amlogic/meson8m2_n100_1G.dtd
index fdbbb4ce8f9a..345ed70bcd88 100755
--- a/arch/arm/boot/dts/amlogic/meson8m2_n100_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_n100_1G.dtd
@@ -111,13 +111,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8m2_n101_512M.dtd b/arch/arm/boot/dts/amlogic/meson8m2_n101_512M.dtd
index ac1cb20ca23c..f2cedcef9ef8 100755
--- a/arch/arm/boot/dts/amlogic/meson8m2_n101_512M.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_n101_512M.dtd
@@ -110,13 +110,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <4>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <3>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd b/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd
index 1f60387f0b87..c9c44e5fe624 100755
--- a/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd
@@ -110,13 +110,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8m2_pxp.dtd b/arch/arm/boot/dts/amlogic/meson8m2_pxp.dtd
index 552b2836b685..c0e8b87666ac 100755
--- a/arch/arm/boot/dts/amlogic/meson8m2_pxp.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_pxp.dtd
@@ -110,13 +110,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/boot/dts/amlogic/meson8m2_skt.dtd b/arch/arm/boot/dts/amlogic/meson8m2_skt.dtd
index 214fd60d14b6..3a4687b2b0f6 100755
--- a/arch/arm/boot/dts/amlogic/meson8m2_skt.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_skt.dtd
@@ -110,13 +110,13 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "clk_level"
     vpu{
-		compatible = "amlogic,vpu";
-		dev_name = "vpu";	
-		status = "ok";
-		clk_level = <7>;
-		/**    	0: 106.25M		1: 127.5M		2: 159.375M
-				3: 182.15M		4: 212.5M		5: 255M
-		    	6: 318.75M		7: 364.3M					 */
+        compatible = "amlogic,vpu";
+        dev_name = "vpu";	
+        status = "ok";
+        clk_level = <7>;
+        /**    0: 106.25M    1: 127.50M    2: 159.375M
+               3: 212.50M    4: 255.00M    5: 283.33M
+               6: 318.75M    7: 425.00M    */
 	};
 	
 /// **************************************************************************************	
diff --git a/arch/arm/mach-meson8/vpu.c b/arch/arm/mach-meson8/vpu.c
index f03eedc0da98..3cb7b1ffd327 100755
--- a/arch/arm/mach-meson8/vpu.c
+++ b/arch/arm/mach-meson8/vpu.c
@@ -11,6 +11,7 @@
 #include <linux/err.h>
 #include <linux/of.h>
 #include <linux/of_device.h>
+#include <linux/delay.h>
 #include <plat/io.h>
 #include <mach/am_regs.h>
 #include <mach/vpu.h>
@@ -63,21 +64,19 @@ typedef struct {
 }VPU_Conf_t;
 
 #define CLK_LEVEL_DFT		4
-#define CLK_LEVEL_MAX		8	//limit max clk to 364M
+#define CLK_LEVEL_MAX		8	//limit max clk to 425M
 static unsigned int vpu_clk_setting[][3] = {
 	//frequency		clk_mux		div
 	{106250000,		1,			7},	//0
 	{127500000,		2,			3},	//1
 	{159375000,		0,			3},	//2
-	{182150000,		3,			1},	//3
-	{212500000,		1,			3},	//4
-	{255000000,		2,			1},	//5
+	{212500000,		1,			3},	//3
+	{255000000,		2,			1},	//4
+	{283333000,		1,			2},	//5
 	{318750000,		0,			1},	//6
-	{364300000,		3,			0},	//7
-	{425000000,		1,			1},	//8
-	{510000000,		2,			0},	//9
-	{637500000,		0,			0},	//10
-	//{850000000,		1,			0},	//11
+	{425000000,		1,			1},	//7
+	{510000000,		2,			0},	//8
+	{637500000,		0,			0},	//9
 };
 
 static unsigned int vpu_clk_vmod[] = {
@@ -195,7 +194,7 @@ static unsigned int get_vpu_clk_level(unsigned int video_clk)
 	video_bw = video_clk + 1000000;
 
 	for (i=0; i<CLK_LEVEL_MAX; i++) {
-		if (video_bw <= vpu_clk_setting[i][0])			
+		if (video_bw <= vpu_clk_setting[i][0])
 			break;
 	}
 	clk_level = i;
@@ -634,23 +633,51 @@ static struct class aml_vpu_debug_class = {
 //*********************************************************//
 
 static void vpu_driver_init(void)
-{	
-	set_vpu_clk(vpu_config.clk_level);
-	
-	//VPU MEM_PD, need to modify
-	aml_write_reg32(P_HHI_VPU_MEM_PD_REG0, vpu_config.mem_pd0);
+{
+    set_vpu_clk(vpu_config.clk_level);
+
+    aml_set_reg32_bits(P_AO_RTI_GEN_PWR_SLEEP0, 0, 8, 1); // [8] power on
+    aml_write_reg32(P_HHI_VPU_MEM_PD_REG0, vpu_config.mem_pd0);
     aml_write_reg32(P_HHI_VPU_MEM_PD_REG1, vpu_config.mem_pd1);
+    aml_set_reg32_bits(P_HHI_MEM_PD_REG0, 0, 8, 8); // MEM-PD
+    udelay(2);
+
+    //Reset VIU + VENC
+    //Reset VENCI + VENCP + VADC + VENCL
+    //Reset HDMI-APB + HDMI-SYS + HDMI-TX + HDMI-CEC
+    aml_write_reg32(P_RESET0_MASK, aml_read_reg32(P_RESET0_MASK) & (~((0x1 << 5) | (0x1<<10))));
+    aml_write_reg32(P_RESET4_MASK, aml_read_reg32(P_RESET4_MASK) & (~((0x1 << 6) | (0x1<<7) | (0x1<<9) | (0x1<<13))));
+    aml_write_reg32(P_RESET2_MASK, aml_read_reg32(P_RESET2_MASK) & (~((0x1 << 2) | (0x1<<3) | (0x1<<11) | (0x1<<15))));
+    aml_write_reg32(P_RESET2_REGISTER, ((0x1 << 2) | (0x1<<3) | (0x1<<11) | (0x1<<15)));
+    aml_write_reg32(P_RESET4_REGISTER, ((0x1 << 6) | (0x1<<7) | (0x1<<9) | (0x1<<13)));    // reset this will cause VBUS reg to 0
+    aml_write_reg32(P_RESET0_REGISTER, ((0x1 << 5) | (0x1<<10)));
+    aml_write_reg32(P_RESET4_REGISTER, ((0x1 << 6) | (0x1<<7) | (0x1<<9) | (0x1<<13)));
+    aml_write_reg32(P_RESET2_REGISTER, ((0x1 << 2) | (0x1<<3) | (0x1<<11) | (0x1<<15)));
+    aml_write_reg32(P_RESET0_MASK, aml_read_reg32(P_RESET0_MASK) | ((0x1 << 5) | (0x1<<10)));
+    aml_write_reg32(P_RESET4_MASK, aml_read_reg32(P_RESET4_MASK) | ((0x1 << 6) | (0x1<<7) | (0x1<<9) | (0x1<<13)));
+    aml_write_reg32(P_RESET2_MASK, aml_read_reg32(P_RESET2_MASK) | ((0x1 << 2) | (0x1<<3) | (0x1<<11) | (0x1<<15)));
+
+    //Remove VPU_HDMI ISO
+    aml_set_reg32_bits(P_AO_RTI_GEN_PWR_SLEEP0, 0, 9, 1); // [9] VPU_HDMI
 }
 
 static void vpu_driver_disable(void)
 {
-	vpu_config.mem_pd0 = aml_read_reg32(P_HHI_VPU_MEM_PD_REG0);
-	vpu_config.mem_pd1 = aml_read_reg32(P_HHI_VPU_MEM_PD_REG1);
-	
-	aml_write_reg32(P_HHI_VPU_MEM_PD_REG0, 0xffffffff);
+    vpu_config.mem_pd0 = aml_read_reg32(P_HHI_VPU_MEM_PD_REG0);
+    vpu_config.mem_pd1 = aml_read_reg32(P_HHI_VPU_MEM_PD_REG1);
+
+    // Power down VPU_HDMI
+    // Enable Isolation
+    aml_set_reg32_bits(P_AO_RTI_GEN_PWR_SLEEP0, 1, 9, 1); // ISO
+    //Power off memory
+    aml_write_reg32(P_HHI_VPU_MEM_PD_REG0, 0xffffffff);
     aml_write_reg32(P_HHI_VPU_MEM_PD_REG1, 0xffffffff);
-	
-	aml_set_reg32_bits(P_HHI_VPU_CLK_CNTL, 0, 8, 1);
+    aml_set_reg32_bits(P_HHI_MEM_PD_REG0, 0xff, 8, 8); // HDMI MEM-PD
+
+    //Power down VPU domain
+    aml_set_reg32_bits(P_AO_RTI_GEN_PWR_SLEEP0, 1, 8, 1); //PDN
+
+    aml_set_reg32_bits(P_HHI_VPU_CLK_CNTL, 0, 8, 1);
 }
 
 #ifdef CONFIG_PM
-- 
2.19.0

